indigo/indigo-extras/src/main/scala/indigoextras/waypoints/WaypointPath.scala

```diff
@@ -0,0 +1,113 @@
+package indigoextras.waypoints
+
+import indigo.Radians
+import indigo.Vector2
+
+import scala.annotation.tailrec
+
+trait WaypointPath:
```

Comment:
Usually I use case classes here to make this easier, but either way we'll probably want method like `def withRadius` that, combined with the comment below about apply methods, allow you to say things like `WaypointPath(v1, v2, v3).withRadius(12)`.

Reply:
Also we're going to need some Scaladocs.

Reply:
I think I started with the trait with idea of making different implementations depending on the underlying interpolator (like linear or bezier or something like that). But it's overcomplicating and I think using a case class is easier indeed. I'll add the builder methods as well.